src/page-modules/assistant/client/journey-planner/index.ts

```diff
@@ -41,14 +46,38 @@ export function useTripPatterns(tripQuery: FromToTripQuery) {
       {},
     );
 
-  const isLoadingInitialData = !data && isLoading;
+  useEffect(() => {
+    if (data) {
+      setNumberOfTripPatterns(
+        data.reduce((acc, curr) => acc + curr.tripPatterns.length, 0),
+      );
+    }
+  }, [data]);
+
+  useEffect(() => {
+    if (isLoading || isValidating) {
+      return;
+    }
+    if (size >= MAX_NUMBER_OF_INITIAL_SEARCH_ATTEMPTS) {
+      return;
+    }
+    if (numberOfTripPatterns >= INITIAL_NUMBER_OF_WANTED_TRIP_PATTERNS) {
+      return;
+    }
+    setSize(size + 1);
+  }, [numberOfTripPatterns, size, setSize, isLoading, isValidating]);
+
+  const isLoadingFirstPage = !data && isLoading;
```

Comment:
This is set to false after the first trip is loaded? Would `isLoadingFirstTrip` be more fitting?

Reply:
Yes, false because we have data at that point. I agree, updated 👍